publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java

```diff
@@ -304,52 +326,40 @@ private Resource sendUpdateRequest(UpdateItemRequest updateRequest) {
             .orElseThrow();
     }
 
-    private UpdateItemRequest publishUpdateRequest(ResourceDao dao) throws JsonProcessingException {
+    private UpdateItemRequest publishUpdateRequest(ResourceDao dao) {
+
+        final String updateExpression = "SET"
+                                        + " #resource.#status = :newStatus, "
+                                        + "#resource.#modifiedDate = :modifiedDate, "
+                                        + "#resource.#publishedDate = :modifiedDate ";
 
-        ConcurrentHashMap<String, String> expressionNamesMap = publishUpdateExpressionNamesMap();
-        ConcurrentHashMap<String, AttributeValue> expressionValuesMap =
-            updateStatusExpresionValuesMap(PublicationStatus.PUBLISHED);
+        final String conditionExpression = "#resource.#status <> :publishedStatus";
+
+        Map<String, String> expressionNamesMap = Map.of(
+            "#resource", RESOURCE_FIELD_IN_RESOURCE_DAO,
+            "#status", STATUS_FIELD_IN_RESOURCE,
+            "#modifiedDate", MODIFIED_FIELD_IN_RESOURCE,
+            "#publishedDate", PUBLISHED_DATE_FIELD_IN_RESOURCE
+        );
+
+        Map<String, AttributeValue> expressionValuesMap = Map.of(
+            ":newStatus", new AttributeValue(PublicationStatus.PUBLISHED.getValue()),
+            ":modifiedDate", new AttributeValue(nowAsString()),
+            ":publishedStatus", new AttributeValue(PublicationStatus.PUBLISHED.getValue()));
 
-        String updateExpression = "SET"
-                                  + " #resource.#status = :status, "
-                                  + "#resource.#modifiedDate = :modifiedDate, "
-                                  + "#resource.#publishedDate = :modifiedDate ";
         return new UpdateItemRequest()
             .withTableName(tableName)
             .withKey(dao.primaryKey())
             .withUpdateExpression(updateExpression)
-            .withConditionExpression("#resource.#status <> " + PublicationStatus.PUBLISHED.toString())
+            .withConditionExpression(conditionExpression)
             .withExpressionAttributeNames(expressionNamesMap)
             .withExpressionAttributeValues(expressionValuesMap)
             .withReturnValues(ReturnValue.ALL_NEW);
     }
 
-    private ConcurrentHashMap<String, String> publishUpdateExpressionNamesMap() {
```

Comment:
wrote the queries as strings for better readability. I find the cognitive cost of understanding the queries to be higher than the benefit of reusing some strings. In case of a refactoring tests will fail and the developer will need to update every query separately as indicated by the tests.